packages/remirror__extension-file/src/file-upload-context.ts

```diff
@@ -0,0 +1,43 @@
+export interface UploadContext {
+  set: (key: string, value: unknown) => void;
+  get: (key: string) => unknown;
+  addListener: (listener: UploadContextListener) => void;
+}
+
+type UploadContextListener = (values: Record<string, unknown>) => void;
+
+export function createUploadContext(): UploadContext {
+  const values: Record<string, unknown> = {};
+  const eventListeners: UploadContextListener[] = [];
+  let timeoutId: ReturnType<typeof setTimeout> | null = null;
+
+  const get = (key: string): unknown => {
+    return values[key];
+  };
+
+  const set = (key: string, value: unknown) => {
+    values[key] = value;
+    afterSet();
+  };
+
+  const addListener = (listener: UploadContextListener) => {
+    eventListeners.push(listener);
+  };
```

Comment:
We're using `nanoevents` [elsewhere in the codebase](https://github.com/remirror/remirror/blob/a2ca7a83f35b3831b97817eb2cb38b1a82d60ab8/packages/remirror__extension-positioner/src/positioner.ts#L299-L304), should we use that here too?

Reply:
Good idea. It does make the code simpler and cleaner.